base/dates/io.jl

```diff
@@ -99,13 +128,14 @@ for (tok, fn) in zip("eE", [dayabbr, dayname])
     end
 end
 
-function fixwidth(num, n)
-    assert(num>=0)
-    dec(num,n)[end-(n-1):end]
-end
-
 function format(io, d::DatePart{'y'}, dt)
-    write(io, fixwidth(year(dt), d.width))
+    y = year(dt)
+    n = d.width
+
+    # the last `n` digits of `y`
+    # will be 0 padded if `num` has less than `n` digits
+    str = dec(num,n)[end-(n-1):end]
```

Comment:
y, not num? latter isn't defined here (does this cause a test failure? if not, needs coverage)